File: 4-WP-web-programming/Exam/Exam Samples/examjsp/src/main/java/ubb/exam/model/Item.java
package ubb.exam.model;

public class Item extends BaseEntity {
    private String name;
    private String description;
    private Integer value;
    private Long userId;

    public Item() {}

    public Item(String name, String description, Integer value, Long userId) {
        this.name = name;
        this.description = description;
        this.value = value;
        this.userId = userId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Integer getValue() {
        return value;
    }

    public void setValue(Integer value) {
        this.value = value;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    @Override
    public String toString() {
        return "Item{id=" + id +
                "name=" + name +
                "description=" + description +
                "value=" + value +
                "userId=" + userId +
                "}";
    }

    @Override
    public boolean equals(Object obj) {
        if (!(obj instanceof Item))
            return false;
        Item other = (Item)obj;
        return this.name.equals(other.getName()) &&
                this.value.equals(other.value);
    }
}

File: 4-WP-web-programming/Exam/Exam Samples/examphp/item.php
<?php
class Item implements JSONSerializable {
    private $item_id;
    private $name;
    private $description;
    private $value;
    private $user_id;

    public function __construct($item_id, $name, $description, $value, $user_id) {
        $this->item_id = $item_id;
        $this->name = $name;
        $this->description = $description;
        $this->value = $value;
        $this->user_id = $user_id;
    }

    public function getId() {
        return $this->item_id;
    }

    public function getName() {
        return $this->name;
    }

    public function getDescription() {
        return $this->description;
    }

    public function getValue() {
        return $this->value;
    }

    public function getUserId() {
        return $this->user_id;
    }

    public function jsonSerialize() {
        return get_object_vars($this);
    }

}
?>